### ISSUE TITLE
Resize behavior

### ISSUE THREAD

**Issue by djvs** [kovidgoyal/kitty#5637]
**Describe the bug**
Not exactly a bug - when resizing on Wayland (sway) there's roughly a 1 second lag, during which the resizing will actually stretch the current image, until the terminal recalculates rows/cols and fills out the new area.  Speaking for myself I heavily depend on rapid feedback to know when the terminal is the right size, like if I'm resizing it to the maximum width of text from the left side, which was the behavior until the most recent update.  Even a frozen image would be preferable considering I would be able to still see how close the edge is to what I'm trying to resize it to.  Is there any way to control this behavior?

**To Reproduce**
See above

**Screenshots**
Don't think I could get one

**Environment details**
```
kitty 0.26.4 created by Kovid Goyal
Linux blu 6.0.2-artix1-1.1 #1 SMP PREEMPT_DYNAMIC Sun, 16 Oct 2022 21:55:53 +0000 x86_64
Artix Linux 6.0.2-artix1-1.1 (/dev/tty)
DISTRIB_ID="Artix"
DISTRIB_RELEASE="rolling"
DISTRIB_DESCRIPTION="Artix Linux"
Running under: Wayland
Frozen: False
Paths:
  kitty: /usr/bin/kitty
  base dir: /usr/lib/kitty
  extensions dir: /usr/lib/kitty/kitty
  system shell: /usr/bin/zsh
Loaded config files:
  /home/d/.config/kitty/kitty.conf

Config options different from defaults:
background_opacity      0.97
bold_font               Inconsolata for Powerline Medium
bold_italic_font        Inconsolata for Powerline Medium Italic
confirm_os_window_close 0
font_family             Inconsolata for Powerline
font_size               10.0
initial_window_height   (500, 'px')
italic_font             Inconsolata for Powerline Italic
modify_font:
    baseline 0
    cell_height 0
    cell_width 0
open_url_with           ['/usr/bin/xdg-open']
remember_window_size    False
scrollback_lines        20000
tab_bar_edge            1
tab_bar_style           powerline
tab_powerline_style     slanted
tab_title_template      {title}
update_check_interval   24.0
url_prefixes            ('http', 'https', 'file', 'ftp', 'gemini', 'irc', 'gopher', 'mailto', 'news', 'git')
url_style               1
Added mouse actions:
	left release grabbed →  mouse_handle_click link
	left release ungrabbed →  mouse_handle_click link
Added shortcuts:
	alt+1 →  goto_tab 1
	alt+2 →  goto_tab 2
	alt+3 →  goto_tab 3
	alt+4 →  goto_tab 4
	alt+5 →  goto_tab 5
	alt+6 →  goto_tab 6
	alt+7 →  goto_tab 7
	alt+8 →  goto_tab 8
	alt+9 →  goto_tab 9
	alt+c →  copy_to_clipboard
	alt+left →  previous_tab
	alt+right →  next_tab
	alt+super+, →  debug_config
	alt+super+page_down →  scroll_line_down
	alt+super+page_up →  scroll_line_up
	alt+super+r →  clear_terminal reset active
	alt+t →  new_tab
	alt+v →  paste_from_clipboard
	alt+w →  close_tab
	ctrl+- →  change_font_size all -2.0
	ctrl+= →  change_font_size all +2.0
	ctrl+super+, →  load_config_file
	ctrl+super+space →  kitten unicode_input
	shift+super+- →  change_font_size all -2.0
	shift+super+= →  change_font_size all +2.0
	shift+super+[ →  previous_tab
	shift+super+] →  next_tab
	shift+super+d →  close_window
	shift+super+i →  set_tab_title
	shift+super+w →  close_os_window
	super++ →  change_font_size all +2.0
	super+, →  edit_config_file
	super+- →  change_font_size all -2.0
	super+0 →  change_font_size all 0
	super+1 →  first_window
	super+2 →  second_window
	super+3 →  third_window
	super+4 →  fourth_window
	super+5 →  fifth_window
	super+6 →  sixth_window
	super+7 →  seventh_window
	super+8 →  eighth_window
	super+9 →  ninth_window
	super+= →  change_font_size all +2.0
	super+down →  scroll_line_down
	super+end →  scroll_end
	super+enter →  new_window
	super+home →  scroll_home
	super+n →  new_os_window
	super+page_down →  scroll_page_down
	super+page_up →  scroll_page_up
	super+r →  start_resizing_window
	super+t →  new_tab
	super+up →  scroll_line_up
	super+w →  close_tab
Colors:
	background              #161616   
	color0                  #073642   
	color1                  #dc322f   
	color10                 #586e75   
	color11                 #657b83   
	color12                 #839496   
	color13                 #6c71c4   
	color14                 #93a1a1   
	color15                 #fdf6e3   
	color2                  #859900   
	color3                  #b58900   
	color4                  #268bd2   
	color5                  #d33682   
	color6                  #229eb9   
	color7                  #eee8d5   
	color8                  #027997   
	color9                  #cb4b16   
	selection_background    #eeeeee   

Important environment variables seen by the kitty process:
	PATH                                /home/d/.rbenv/shims:/home/d/.rbenv/bin:/tmp/ycm/fnm_multishells/3114_1666910135335/bin:/home/d/.fnm:/opt/google-cloud-sdk/bin:/home/d/.config/composer/vendor/bin:/usr/local/heroku/bin:/home/d/.nvm/versions/node/v10.18.0/bin/:/home/d/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/d/.local/bin/:/home/d/bin/:/opt/npm/bin:/usr/share/elasticsearch/bin:/opt/play/activator-dist-1.3.10/bin:/usr/share/java/apache-ant/bin/ant:/opt/plan9/bin:/opt/torcs/lib/torcs/lib:/home/d/.cabal/bin:/opt/radrails:/home/d/.gem/ruby/1.9.1/bin:/usr/games:/home/d/nodejs/bin:/home/d/.dropbox-dist/:/opt/zed/zed:/opt/viber:/bin:/opt/jre1.8.0_141/bin:/home/d/.local/bin/:/opt/aws/:/home/d/.pulumi/bin/:/usr/share/java/apache-ant/bin/ant:/opt/plan9/bin:/opt/torcs/lib/torcs/lib:/home/d/.cabal/bin:/opt/radrails:/home/d/.gem/ruby/1.9.1/bin:/usr/share/viber:/opt/android/adt-bundle-linux-x86_64-20140702/sdk/platform-tools:/opt/android/adt-bundle-linux-x86_64-20140702/sdk/tools:/opt/android/adt-bundle-linux-x86_64-20140702/eclipse:/opt/android-sdk/emulator:/opt/monero-gui-v0.14.0.0:/home/d/.local/bin:/home/d/.nvm/versions/node/v8.9.4/bin/:/home/d/.local/rtorrent/0.9.6-PS-1.2-dev/bin/:/opt/helm/linux-amd64:/opt/postman:/opt/android-sdk/tools:/opt/android-sdk/platform-tools
	LANG                                en_US.UTF-8
	VISUAL                              nvim
	EDITOR                              nvim
	SHELL                               /usr/bin/zsh
	DISPLAY                             :0
	WAYLAND_DISPLAY                     wayland-1
	USER                                d
	XCURSOR_SIZE                        24
	XDG_SEAT                            seat0
	XDG_SESSION_TYPE                    wayland
	XDG_SESSION_CLASS                   user
	XDG_VTNR                            1
	XDG_SESSION_ID                      1
	XDG_RUNTIME_DIR                     /run/user/1000
	XDG_DATA_DIRS                       /home/d/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share

```

**Additional context**

Still happens with kitty --config NONE.

**Comment by kovidgoyal** [comment]
dup of #5601

**Comment by raboof** [comment]
(sorry to comment on such an old issue, but for those who come here via web search like me: a very similar issue seems to have appeared in 0.35.1 and was fixed again in 0.35.2)
